src/main/java/de/tum/cit/aet/artemis/core/service/FileService.java

```diff
@@ -186,6 +186,32 @@ public URI handleSaveFile(MultipartFile file, boolean keepFilename, boolean mark
         return URI.create(markdown ? MARKDOWN_FILE_SUBPATH : DEFAULT_FILE_SUBPATH).resolve(currentFilename);
     }
 
+    /**
+     * Handles the saving of a file in a conversation.
+     *
+     * @param file           The file to be uploaded.
+     * @param courseId       The ID of the course.
+     * @param conversationId The ID of the conversation.
+     * @return The URI of the saved file.
+     */
+    public URI handleSaveFileInConversation(MultipartFile file, Long courseId, Long conversationId) {
+        // TODO: Improve the access check. The course is already checked, but the user might not be a member of the conversation. The course may not belong to the conversation
+        String filename = checkAndSanitizeFilename(file.getOriginalFilename());
+
+        validateExtension(filename, true);
```

Comment:
The server and client both implement a check for this. If used normally, the client should never get a bad request error from the server because both the server and client store the same set of extensions (de.tum.cit.aet.artemis.core.service.FileService#allowedMarkdownFileExtensions and file-extensions.constants.ts#UPLOAD_MARKDOWN_FILE_EXTENSIONS) and the client will not perform the request:

```ts
        if (!this.validateFileExtension(file)) {
            return Promise.reject(
                new Error(`Unsupported file type! Only the following file extensions are allowed: ${this.acceptedMarkdownFileExtensions.map((ext) => `.${ext}`).join(', ')}.`),
            );
        }
```